### DISCUSSION TITLE
Pull Nodes from Robot

### DISCUSSION THREAD

**Discussion by gabrielyakoub** [BHoM/BHoM#1439] [Q&A] [answered]
Hi, I wonder if there is a way to pull specific nodes from Robot instead of pulling all the nodes in the model. Thanks in advance!

**Comment by FraserGreenroyd** [comment]
@IsakNaslundBh are you able to assist here?

**Comment by IsakNaslundBh** [comment, marked as answer]
Hi @gabrielyakoub , yes there is.

See attached script for either pulling nodes with a set ID, or nodes selected in the model.

The pulling by ids could and probably should be made more clear and distinct, but what is in the script should work for now.

Hope this helps :)

[Filter Nodes pull Robot.zip](https://github.com/BHoM/BHoM/files/9828937/Filter.Nodes.pull.Robot.zip)

**Reply by gabrielyakoub** [reply]
Thanks a lot @IsakNaslundBh, it was really helpful. :)

**Reply by MMabruk** [reply]
Hi @IsakNaslundBh

This will help make my scripts more efficient and quicker to run, because rather than pulling all nodes in a Robot model and then post-processing, filtering from the start specific robot ID's means pulling less data. However, there is still a manual step which involves identifying in Robot the nodes you want to pull, copying their Robot ID's into the text panel in grasshopper. 

Is it possible at all to filter by other properties? Nodes, with specific support conditions, or coordinates (z=0 for example). Or Bars with specific names, section sizes, or releases? There are so many properties you can isolate for nodes and bars, just wondering if it's possible to do that using BHoM to limit the amount of objects pulled from the start?

Thanks

**Reply by IsakNaslundBh** [reply]
> Is it possible at all to filter by other properties? Nodes, with specific support conditions, or coordinates (z=0 for example). Or Bars with specific names, section sizes, or releases? There are so many properties you can isolate for nodes and bars, just wondering if it's possible to do that using BHoM to limit the amount of objects pulled from the start?

Hi @MMabruk ,

There is currently nothing like what you described implemented in the Robot_Adapter. Most of the above could be added (especially relating to filtering by different labels, such as Supports, Releases, SectionSizes etc), and could have a look into how to enable that.

Have raised an issue for this here: https://github.com/BHoM/Robot_Toolkit/issues/499 .
Cannot promise any time line when this can be tackled though.

Filtering based on geometric position (such as z=0) is not something supported by the Robot API, and hence would require to be filtered via post-processing which could defy the purpose a bit, as the most computationally expensive part of pulling from Robot is the com calls to the API. Ofc, it is something that also could be added, but given it will run about as quick post-processing it in something like Grasshopper, it will probably not be given the highest priority atm.
